build_runner/lib/src/generate/phase.dart

```diff
@@ -7,46 +7,59 @@ import 'package:build/build.dart';
 import '../package_builder/package_builder.dart';
 import 'input_set.dart';
 
-/// A "phase" in the build graph, which represents running a [builder] on a
+/// A "phase" in the build graph, which represents running a builder on a
 /// specific [package].
+///
+/// See the [BuildAction] and [PackageBuildAction] implementations.
 abstract class BuildAction {
   String get package;
 
+  /// If `true` then this action runs lazily when another action tries to read
```

Comment:
[nit] "Whether to run lazily when another action..."

Also a blank line following the first sentence. Maybe:

"Whether to run lazily when an output is read.

An optional builder will only run if one of it's outputs is read by a later Builder, or is used as a primary input to a Builder. If no build actions read the output of an optional builder it will never run."

Reply:
done